### PR TITLE
Login/Logout along with Framework of Registation

### PR DESCRIPTION
**Pull request by oetago**
Added User Login and Logout
Added framework for User Registration - Dialog and QB Sign up method
Changed Login layout to Fragment
Added Interfaces for inter Fragment communication
Added start Session of QB Application

### CODE REVIEW

**Comment by oetago** [comment]
+1

File: brewchat/src/main/java/com/example/brewchat/activities/ChatManagerActivity.java [hunk 1]
```diff
@@ -50,4 +53,29 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    @Override
+    public void onBackPressed() {
+        AlertDialog.Builder alertDialog = new AlertDialog.Builder(this);
+        alertDialog.setTitle("Logout"); // Sets title for your alertbox
```
Comment (arreche) [inline brewchat/src/main/java/com/example/brewchat/activities/ChatManagerActivity.java]: `Logout` caption can be moved to the strings file

**Comment by arreche** [comment]
+1

Butterknife provides a handy @OnClick annotation for clicklisteners
See http://jakewharton.github.io/butterknife/

**Comment by jephillips** [comment]
+1

**Comment by oetago** [comment]
+1

**Pull request merged**